library/std/src/sys/uefi/time.rs

```diff
@@ -0,0 +1,102 @@
+//! Time is implemeted using `EFI_RUNTIME_SERVICES.GetTime()`
+//! While this is  not technically monotonic, the single-threaded nature of UEFI might make it fine
+//! to use for Instant. Still, maybe revisit this in future.
```

Comment:
Since the time may be adjusted with `SetTime()`, monotonicity can really not be relied upon here. Perhaps the CPU counter registers (the TSC on x86_64 and the EL1 timer on Aarch64) are a better choice?

Reply:
How about using `EFI_TIMESTAMP_PROTOCOL` (defined in Section 38.1 in UEFI Specification)?

Reply:
That is perfect, but it unfortunately might not be supported (it was added in 2013 with UEFI 2.4 and is not mandatory to implement).

Reply:
Yeah. We can use it if it is found. If it is missing, then use the counter registers?

Reply:
@joboet Ok, so after a lot of searching, I have some idea how `Instant` can be implemented using counter registers. However, it seems quite complicated to do.
The main problem is, that I do not think UEFI has any way to get the TSC counter frequency. Without that, it's not possible to convert the register value into some kind of time. The Linux kernel obtains this frequency value by using [`calibrate_tsc()`](https://github.com/torvalds/linux/blob/568035b01cfb107af8d2e4bd2fb9aea22cf5b868/arch/x86/kernel/tsc.c#L622). So it will need a lot of code to make `Instant` work.

Reply:
Resolving this conversation now. Feel free to open new conversations about improving the TSC implementation (since this is my first time working with Assembly, there might be problems).